### PR TITLE
Fix for `WOLFSSL_ALT_CERT_CHAINS` with long chain

### PR DESCRIPTION
**Pull request by dgarske**
* Fix for `WOLFSSL_ALT_CERT_CHAINS` incorrectly failing on success case.
* Fix for `NO_WOLFSSL_SERVER` typo.

* Cleanup of the myVerify callback to allow specific actions.
* Fix the suites.c to not crash if no newline at end of file or extra newlines in .test files
* Added helpful debug message to show that a CA was found.
* Added second intermediate CA to testing certs. This creates a chain that looks like: `ROOT (www.wolfssl.com) -> INT (wolfSSL Intermediate CA) -> INT2 (wolfSSL Intermediate2 CA) -> PEER (wolfSSL Client Chain / wolfSSL Server Chain).`
* Add alternate chain test case where peer presents chain (INT -> INT2 -> PEER) and only INT2 is loaded as trusted.

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -10855,7 +10855,8 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                                 ssl->options.usingAltCertChain = 1;
                             }
 
-                            ret = 0; /* clear error and continue */
+                            ret = 0; /* clear errors and continue */
+                            args->verifyErr = 0;
```
Comment (toddouska) [inline src/internal.c:10859]: Can you add a test case for this?

Also, are there any potential verifyErrs we don't want to override here?

Comment (dgarske) [reply src/internal.c:10859]: @toddouska : Will do on the test case. The case is a chain of at least 3 certs where the intermediate is loaded as trusted and  peer certificate was validated correctly, but the intermediate failed with -188 (ASN_NO_SIGNER_E) error in `args->verifyErr`, which incorrectly throws the error. I think to be safe here we could just override `ASN_NO_SIGNER_E`, `ASN_BEFORE_DATE_E` and `ASN_AFTER_DATE_E`.

Comment (dgarske) [reply src/internal.c:10852]: @toddouska : For reference the only error that gets overridden is `ASN_NO_SIGNER_E`. The test case for this requires ROOT CA -> INT CA -> INT CA2 -> PEER where only INT CA2 is loaded as trusted. I'll build up a test chain for this and add the test case.

Comment (dgarske) [reply src/internal.c:10958]: @toddouska : Pushed some updated tests. Turns out this fix was only encountered when using the verify callback with alternate cert chains enabled with a long chain. The non-verify callback cases were handled correctly. So I updated the myVerify callback to allow for some more options.

**Pull request merged**